File: solutions/typescript/leap/1/leap.ts
export function isLeap(year: number): boolean {
  let isLeapYear: boolean = false;
  if(year%4==0 && year%100!==0){
    isLeapYear = true;
  }
  else if(year%4==0 && year%100==0 && year%400==0){
    isLeapYear = true;
  }
  return isLeapYear;
}

File: solutions/java/leap/1/src/main/java/Leap.java
class Leap {
    boolean isLeapYear(int year) {
        boolean isDivBy4 = year % 4 == 0;
        boolean isDivBy100 = year % 100 == 0;
        boolean isDivBy400 = year % 400 == 0;
        boolean isLeapYear = (isDivBy4 && !isDivBy100 || isDivBy4 && isDivBy100 && isDivBy400);
        return isLeapYear;
    }
}
